src/main/java/com/couchbase/lite/android/AndroidSQLCipherStorageEngine.java

```diff
@@ -22,89 +22,94 @@
 import com.couchbase.lite.storage.SQLiteStorageEngine;
 import com.couchbase.lite.util.Log;
 
-import net.sqlcipher.database.SQLiteDatabase;
-//import net.sqlcipher.database.SQLiteException;
+import com.couchbase.lite.database.sqlite.SQLiteDatabase;
+
+import java.util.Map;
 
 public class AndroidSQLCipherStorageEngine implements SQLiteStorageEngine {
-    public static final String TAG = "AndroidSQLCipherStorageEngine";
+    private static final boolean SUPPORT_ENCRYPTION = true;
 
     private SQLiteDatabase database;
-    private AndroidContext context;
 
-    public AndroidSQLCipherStorageEngine(AndroidContext context) {
-        this.context = context;
+    public AndroidSQLCipherStorageEngine() {
+        // Do Nothing
     }
 
     @Override
-    public boolean open(String path, String encryptionKey) throws SQLException {
+    public boolean open(String path) throws SQLException {
         if(database != null && database.isOpen())
             return true;
-
         try {
-            // Write-Ahead Logging (WAL) http://sqlite.org/wal.html
-            // http://developer.android.com/reference/android/database/sqlite/SQLiteDatabase.html#enableWriteAheadLogging()
-            // ENABLE_WRITE_AHEAD_LOGGING is available from API 16
-            // enableWriteAheadLogging() is available from API 11, but it does not work with API 9 and 10.
-            // Minimum version CBL Android supports is API 9
-
-            // NOTE: Not obvious difference. But it seems Without WAL is faster.
-            //       WAL consumes more memory, it might make GC busier.
-            SQLiteDatabase.loadLibs(context.getWrappedContext());
-            char[] key = encryptionKey != null ? encryptionKey.toCharArray() : new char[0];
-            database = SQLiteDatabase.openOrCreateDatabase(path, key, null);
+            database = SQLiteDatabase.openDatabase(path, null, SQLiteDatabase.CREATE_IF_NECESSARY);
             Log.v(Log.TAG_DATABASE, "%s: Opened Android sqlite db", this);
-
-            String sqliteDatabaseClassName = "net/sqlcipher/database/SQLiteDatabase";
-            SQLiteJsonCollator.register(database, sqliteDatabaseClassName, 0);
-            SQLiteRevCollator.register(database, sqliteDatabaseClassName, 0);
-        } catch(net.sqlcipher.database.SQLiteException e) {
-            Log.e(TAG, "Unable to open the SQLite database", e);
+        } catch(Throwable e) {
+            Log.e(Log.TAG_DATABASE, "Unable to open the SQLite database", e);
             if (database != null)
                 database.close();
             throw new SQLException(e);
         }
-
         return database.isOpen();
     }
 
     @Override
     public int getVersion() {
```

Comment:
This is general comments for all methods that are declared in this class.

As all method could throw SQLException, but method signature is not defined with throws SQLException. Is it correct?